hail/python/hailtop/hailctl/config/initialize.py

```diff
@@ -0,0 +1,240 @@
+                                       default=True)
+    if create_remote_tmpdir:
+        remote_tmpdir, warnings = await setup_new_remote_tmpdir(
+            bucket_name=maybe_bucket_name,
+            region=region,
+            project=project,
+            username=username,
+            service_account=hail_identity,
+            verbose=verbose
+        )
+    else:
+        remote_tmpdir, warnings = await setup_existing_remote_tmpdir(region, hail_identity, verbose)
+
+    return (remote_tmpdir, region, warnings)
+
+
+async def async_basic_initialize(incremental: bool = False, verbose: bool = False):
+    from hailtop.auth.auth import async_get_userinfo  # pylint: disable=import-outside-toplevel
+    from hailtop.batch_client.aioclient import BatchClient  # pylint: disable=import-outside-toplevel
+    from hailtop.config import DeployConfig  # pylint: disable=import-outside-toplevel
+    from hailtop.config import get_user_config_path  # pylint: disable=import-outside-toplevel
+
+    from ..auth.login import async_login  # pylint: disable=import-outside-toplevel
+    from .cli import set as set_config, list as list_config  # pylint: disable=import-outside-toplevel
+    from .utils import already_logged_into_service, get_default_region  # pylint: disable=import-outside-toplevel
+
+    domain = Prompt.ask('What domain is the Hail service running in?', default='hail.is')
+    namespace = 'default'
+
+    deploy_config = DeployConfig('external', namespace, domain)
+    deploy_config.dump_to_file()
+
+    already_logged_in = await already_logged_into_service()
+    if not already_logged_in:
+        await async_login(namespace)
+
+    user_info = await async_get_userinfo()
+    username = user_info['username']
+    hail_identity = user_info['hail_identity']
+    trial_bp_name = user_info['trial_bp_name']
+
+    batch_client = await BatchClient.create(trial_bp_name, deploy_config=deploy_config)
+
+    async with batch_client:
+        cloud = await batch_client.cloud()
+        supported_regions = await batch_client.supported_regions()
+        default_region = get_default_region(supported_regions, 'gcp')
+
+        if cloud == 'gcp':
+            remote_tmpdir, region, warnings = await initialize_gcp(username, hail_identity, supported_regions, default_region, verbose)
+        else:
+            region = Prompt.ask('Which region should resources be created in?', default=default_region,
+                                choices=supported_regions)
+
+            remote_tmpdir = Prompt.ask(f'Enter a path to an existing remote temporary directory (ex: gs://my-bucket/batch/tmp)')
+            typer.secho(f'WARNING: You will need to grant read/write access to {remote_tmpdir} for account {hail_identity}', fg=typer.colors.MAGENTA)
+            warnings = True
+
+    config_file = get_user_config_path()
+
+    if not incremental:
+        if os.path.isfile(config_file):
+            os.remove(config_file)
+
+    set_config('domain', domain)
```

Comment:
I think it's a mistake to have the domain in the user config. It's duplicated in the deploy config, not self-explanatory, and it's not something that a user can change without also re-authenticating.

Off the top of my head, I think it would be better if `hailctl auth login hail.is` set the domain in the deploy config and achieved both at once. But regardless of what we end up doing, that's not something that should get too entangled in this PR. I think we should not include setting it in this PR and make a GitHub issue describing how this config variable is duplicated and needs resolution.

Reply:
LMK if you want me to create the issue.

Reply:
I cannot figure out how this is all supposed to work. The deploy config reads the domain value from the `get_user_config()` output if no domain is specified or the environment variable. But running `hailctl auth login hail.is` doesn't take the domain right now. It only takes the namespace.

Can you be more specific on what changes you want in this PR?

Reply:
My apologies, you're right `hailctl auth login hail.is` is just not a thing. My main point here is that I think having `domain` in the user config is a mistake, and I would like not to pollute users configs with this if we are going to change it down the line. I'll make a separate issue detailing my concerns around `domain`, but for this PR I think we should:

1. Not set the domain anywhere. Most (or basically all?) users never set this and it defaults to `hail.is`, so let's just leave that out for now
2. Expect that before this command the user has already logged in.

Reply:
I've long been mystified by users who had `global/config` set, but now I see why. I apparently even approved the change back in [December 2021](https://github.com/hail-is/hail/pull/11113)! I agree with you Daniel, seems odd to store that in two places.

That said, I think this conversation is entirely independent of this PR. If we want to configure the domain in our setup command, that seems entirely reasonable, but let's mutate the deploy_config file, not the user config.